### ISSUE TITLE
Consistent server crash on water tick when water falls into new vertical chunk

### ISSUE THREAD

**Issue by MonzUn** [StrangeLoopGames/EcoIssues#24701]
One of the players on MightyMoose (previously EcoWorld) managed to crash the server three times by placing a pipe with water flow in a specific location. I did some quick debugging and it's because the water flows into a new chunk and it fails to fetch the chunk below and then dereferences the position of that null chunk reference.

Other code seems to have handling for when the chunk comes back as null, but not water ticking.
WaterBlock.cs::60 has the issue and line 62 triggers the nullref

World download:
https://drive.google.com/file/d/1pkIaJiltJDc3gCr_-m0Oc3WpQdtiHpBq/view?usp=sharing

STR:
1. Load the save game from the above link.
2. Go to (600,90,411) and position the pipe as in the picture below.

![image](https://github.com/StrangeLoopGames/EcoIssues/assets/11805935/313933b0-412e-453e-ae8e-9505a13c69e0)

**Comment by padmanek** [comment]
We had the pleasure to get the same crash yesterday:

```
[04:42:49.055] [227] [Error] [Eco] System.AggregateException: One or more errors occurred. (Object reference not set to an instance of an object.)
 ---> System.NullReferenceException: Object reference not set to an instance of an object.
   at Eco.World.World.GetBlockFromChunkIfPossible(Vector3i pos, PersistentChunk chunk)
   at Eco.World.Blocks.WaterBlock.Tick(Vector3i worldPosition, PersistentChunk chunk, PoolableListWrapper`1 blockChanges)
   at Eco.World.PersistentChunk.Tick()
   at Eco.World.WorldPlugin.TickAll(CancellationToken cancellationToken)
   --- End of inner exception stack trace --- System.NullReferenceException: Object reference not set to an instance of an object.
   at Eco.World.World.GetBlockFromChunkIfPossible(Vector3i pos, PersistentChunk chunk)
   at Eco.World.Blocks.WaterBlock.Tick(Vector3i worldPosition, PersistentChunk chunk, PoolableListWrapper`1 blockChanges)
   at Eco.World.PersistentChunk.Tick()
   at Eco.World.WorldPlugin.TickAll(CancellationToken cancellationToken)
```
Crashes at the same piece of code as Monzun's server.

**Comment by D3nnis3n** [comment]
Fixed in 10.1.3.
